neurovault/apps/statmaps/views.py

```diff
@@ -650,6 +658,68 @@ def atlas_query_voxel(request):
         return JSONResponse('error: one or more coordinates are out of range', status=400)
     return JSONResponse(data)
 
+# Compare Two Images
+def compare_images(request,pk1,pk2):
+    image1 = get_image(pk1,None,request)
+    image2 = get_image(pk2,None,request)
+    
+    # Name the data file based on the new volumes
+    path, name1, ext = split_filename(image1.file.url)
+    path, name2, ext = split_filename(image2.file.url)
+
+    # For now, manually choose an atlas - this can be user choice
+    neurovault_static = os.path.dirname(os.path.realpath(neurovault.apps.statmaps.models.__file__))
+    atlas_file = os.path.abspath(os.path.join(neurovault_static, 'static/atlas/MNI-maxprob-thr25-2mm.nii'))
+    atlas_xml = os.path.abspath(os.path.join(neurovault_static, 'static/atlas/MNI.xml'))
+    atlas = pybrainatlas.atlas(atlas_xml,atlas_file) # Default slices are "coronal","axial","sagittal"
+
+    # Create custom image names and links for the visualization
+    custom = {"IMAGE_1":"%s : %s [%s]" %(image1.name,image1.collection.name,image1.map_type),"IMAGE_2": "%s : %s [%s]" %(image2.name,image2.collection.name,image2.map_type),"IMAGE_1_LINK":"/images/%s" %(image1.pk),"IMAGE_2_LINK":"/images/%s" %(image2.pk)}
+
+    html_snippet,data_table = compare.scatterplot_compare(image1=image1.file.path,image2=image2.file.path,software="FREESURFER",voxdim=[8,8,8],atlas=atlas,custom=custom,corr="pearson")
+
+    html = [h.strip("\n") for h in html_snippet]
+    context = {'html': html}
+    return render(request, 'statmaps/compare_images.html', context)
+
+# Return search interface for one image vs rest
+def find_similar(request,pk):
+    image1 = get_image(pk,None,request)
+    pk = int(pk)    
+
+    # Get all similarity calculations for this image, and ids of other images
+    comparisons = Comparison.objects.filter(Q(image1=image1) | Q(image2=image1))    
+    scores = [comp.similarity_score for comp in comparisons]
+    image_ids = list(); image_ids.append(pk)
```

Comment:
@infocortex what's the status?

If an Image gets deleted all objects referring to it will also be deleted (due to ForeignKey mechanism) automatically (at least in theory).

This does not apply to update so you need to rerun comparisons if an Image gets updated with a new file.